### PR TITLE
Add a focusring to page navigation buttons

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #38135 .

### Summary of Changes
As title says

### Testing Instructions
Attention: testing needs npm run  build:css

Open an article in a blog where page navigation is enabled.

### Actual result BEFORE applying this Pull Request
Navigation buttons without focus ring.

### Expected result AFTER applying this Pull Request
Focus is visible

grafik

### Documentation Changes Required
no

### CODE REVIEW

File: build/media_source/templates/site/cassiopeia/scss/blocks/_global.scss [hunk 1]
```diff
@@ -253,6 +253,11 @@ meter {
   }
 }
 
+/* margin for focus ring */
+.page-link {
+    margin: 0.25rem;;
```
Comment (richard67) [inline build/media_source/templates/site/cassiopeia/scss/blocks/_global.scss]:
```suggestion
  margin: .25rem;
```
Fix scss code style failing in drone.

**Comment by ChristineWk** [comment]
Loaded NPM package to test PR. 
After there were changes in the code, of course I needed the new NPM package.
Usually (before new NPM) I load a Nightly, whatever works. You can see the menus above/below, but the page is broken.

Message: Call to undefined method Joomla\Plugin\Behaviour\Versionable\Extension\Versionable::setApplication() 

Can I find this plugin in DB to delete it?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38140">issues.joomla.org/tracker/joomla-cms/38140</a>.</sub>

**Comment by brianteeman** [comment]
go to administrator/cache
delete the file autoload_psr4.php
reload the web page and the file will be rebuilt correc;tly

**Comment by ChristineWk** [comment]
Thanks @brianteeman but:

Had similar in the past (another PR) - the one with the autoload_psr4.php file. Am almost sure it was a different error message - the deletion and reload was successful at that time.

Now: Had successfully deleted. Newly invoked. But still the message. Front end and back end.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38140">issues.joomla.org/tracker/joomla-cms/38140</a>.</sub>

**Comment by joomdonation** [comment]
@ChristineWk Could be because your installation, for some reasons, does not contain the changes from this PR https://github.com/joomla/joomla-cms/pull/38060 . So I guess the easiest way would be download and install 4.2 nightly build to get rid of the error.

**Comment by ChristineWk** [comment]
@joomdonation: Thanks for investigating.

Am sure the changes of PR 38060 were not included. Because: Think it was unfortunately J 4.1.6-dev, not 4.2.
I can no longer go to the backend. 

That means I can only try to install a J 4.2 Nightly over it via FTP and then try to repair the DB.

**Comment by joomdonation** [comment]
OK @ChristineWk . I'm unsure if it works but hope you can get the site recovered.

**Comment by ChristineWk** [comment]
@joomdonation 

Will try it tomorrow. Don't know at the moment. I haven't dared yet. 

Question: Can the reason be that the NPM package included this PR? #38060?

**Comment by joomdonation** [comment]
@ChristineWk I'm unsure. I only guessed the problem base on the error message. It says that setApplication method does not exist and that method was introduced in the mentioned PR

You might want to try to replace the file libraries/src/Plugin/CMSPlugin.php in your Joomla installation with the code from 4.2 branch https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Plugin/CMSPlugin.php to see if it solves the issue. I'm afraid not, but could try.

**Comment by ChristineWk** [comment]
a) First I tried swapping file libraries/src/Plugin/CMSPlugin.php. Not successful.

b) 4.2 Nightly Fully installed/unpacked/loaded etc via FTP.
Of course without /installation, but keeping configuration.php.
Took some hours (Server not fast ...)
The result was that I got the (known) famous "whoops, something went wrong error 500 for hours ...."

c) Then I checked and tried many things, etc. I can't describe everything now....

d) Then I got this message (again):
Class 'Joomla\Plugin\Behavior\Taggable\Extension\Taggable' not found

e) administrator/cache: deleted the autoload_psr4.php. = OK.

Status present: 
Frontend: menus rows are shown, but:
Call to undefined method Joomla\Component\Tags\Site\Helper\RouteHelper::getComponentTagRoute()

Backend: (Instead of Login Mask): Unsatisfied dependency "hotkeys.js" for an asset "script" of type "script"

So, the above was just the short version....
Maybe I'll activate the call stack (via DB) later, .... don't know yet. Is too much for me now.
At the moment, guess I give up...

**Comment by richard67** [comment]
@ChristineWk The PR has been updated meanwhile, and new packages have been built by drone, so maybe you can give it again a try.

**Comment by ChristineWk** [comment]
@richard67 had to completely delete my test page on 30.06. (after many various checks etc. ). will think about it with new 4.2-dev...

**Comment by ChristineWk** [comment]
Tried again with NPM Package:

npm-04jul2022

No focus ring.

npm-04jul2022-2

Database Version shows: 4.2.0-2022-06-22

**Comment by brianteeman** [comment]
You have used your keybpard to tab to the navigation buttoons?

**Comment by ChristineWk** [comment]
When you click, focus flickers short. = OK.

With the tab: a bit different than the screenshot in instructions above. is that correct now?
npm-04jul2022-mit-tab

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 45e4048274d94fec57c8082ba4011123c996f797<br /><br />same results as @christinewk which is not the intention of this PR<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38140">issues.joomla.org/tracker/joomla-cms/38140</a>.</sub>

**Comment by brianteeman** [comment]
The problem is because of the scss in bootstrap _pagination.scss
`  &:focus {
    z-index: 3;
    color: $pagination-focus-color;
    background-color: $pagination-focus-bg;
    outline: $pagination-focus-outline;
    box-shadow: $pagination-focus-box-shadow;
  }
`

As you can see there is a box-shadow being set here and this PR just adds another box-shadow on a different class. Then because of the z-index you get the reported issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38140">issues.joomla.org/tracker/joomla-cms/38140</a>.</sub>

**Comment by chmst** [comment]
Thanks for testing. Will have to work on it when I have time and close for now.

**Comment by richard67** [comment]
I’ve reopened the issue.

**Pull request closed without merging**